graylog2-server/src/main/java/org/graylog2/contentpacks/ContentPackInstallationPersistenceService.java

```diff
@@ -90,4 +94,25 @@ public int deleteById(ObjectId id) {
         final WriteResult<ContentPackInstallation, ObjectId> writeResult = dbCollection.removeById(id);
         return writeResult.getN();
     }
+
+    public Map<ModelId, ContentPackMetaData> getInstallationMetaData() {
```

Comment:
This creates the following structure in the `ContentPackList`:

```json
 "content_packs_meta_data": {
    "1794d39d-077f-7360-b92b-95411b05fbce": {
      "installation_count": {
        "1": 1
      }
    }
  }
```

I think it would be better if we have one metadata object for each content pack revision instead of one metadata object per content pack.
So it would look like this:

```json
 "content_packs_meta_data": {
    "1794d39d-077f-7360-b92b-95411b05fbce": {
      "1": {
         "installation_count": 1
      }
    }
  }
```

Otherwise we would need duplicate the revision for future metadata values.

Reply:
I think the name of this method should be `getContentPackMetadata()` because it returns a map of metadata objects.